pkg/annotations/annotations_test.go

```diff
@@ -645,3 +645,36 @@ func TestExtractResponseBuffering(t *testing.T) {
 		})
 	}
 }
+
+func TestExtractHostAliases(t *testing.T) {
+	type args struct {
+		anns map[string]string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []string
+	}{
+		{
+			name: "empty",
+			want: nil,
+		},
+		{
+			name: "non-empty",
+			args: args{
+				anns: map[string]string{
+					"konghq.com/host-aliases": "foo.kong.com,bar.kong.com",
+				},
+			},
+			want: []string{"foo.kong.com", "bar.kong.com"},
+		},
```

Comment:
Related to the above comment, we can add a small test case when the annotation is misconfigured as an empty string:

```suggestion
		},
		{
			name: "misconfigured",
			args: args{
				anns: map[string]string{
					"konghq.com/host-aliases": "",
				},
			},
			want: nil,
		},
```